sway/input/input-manager.c

```diff
@@ -263,6 +264,32 @@ static void handle_new_input(struct wl_listener *listener, void *data) {
 	input_device->device_destroy.notify = handle_device_destroy;
 }
 
+static void handle_inhibit_activate(struct wl_listener *listener, void *data) {
+	struct sway_input_manager *input_manager = wl_container_of(
+			listener, input_manager, inhibit_activate);
+	struct sway_seat *seat;
+	wl_list_for_each(seat, &input_manager->seats, link) {
+		seat_set_exclusive_client(seat, input_manager->inhibit->active_client);
+	}
+}
+
+static void handle_inhibit_deactivate(struct wl_listener *listener, void *data) {
+	struct sway_input_manager *input_manager = wl_container_of(
+			listener, input_manager, inhibit_deactivate);
+	struct sway_seat *seat;
+	wl_list_for_each(seat, &input_manager->seats, link) {
+		seat_set_exclusive_client(seat, NULL);
+		struct sway_container *previous = seat_get_focus(seat);
+		if (previous) {
+			wlr_log(L_DEBUG, "Returning focus to %p %s '%s'", previous,
+					container_type_to_str(previous->type), previous->name);
+			// Hack to get seat to re-focus the return value of get_focus
```

Comment:
what

Reply:
have you considered using the `send_focus` function?

Reply:
it may be better, set focus to `NULL`.

Reply:
I'm not 100% on the best way to refactor this, I would appreciate it if you filed a follow-up patch improving it if you're not happy with it